server/query.go

```diff
@@ -146,20 +147,39 @@ func ParseQuery(query string, globalRegex bool) (pb.Query, error) {
 		justGotSpace = (match == " ")
 	}
 
+	// This is a special case to provide a better error message,
+	// since the main search term is represented by the "" op.
+	if len(ops[""]) > 1 {
+		return out, fmt.Errorf("main search term must be contiguous")
+	}
+
+	// Handle synonyms
+	out.File = append(ops["file"], ops["path"]...)
+	out.NotFile = append(ops["-file"], ops["-path"]...)
```

Comment:
This _does_ mean we might reorder terms, and the one that ends up in `Line` for a `FilenameOnly` search might not be the first one. I don't know if this matters that much, though :P